packages/devtools_app/lib/src/service/service_manager.dart

```diff
@@ -131,17 +152,42 @@ class ServiceConnectionManager extends ServiceManager<VmServiceWrapper> {
     serviceTrafficLogger?.dispose();
     preferences.vmDeveloperModeEnabled
         .removeListener(_handleVmDeveloperModeChanged);
+  }
 
-    super.afterCloseVmService();
+  Future<void> _handleVmDeveloperModeChanged() async {
+    final vm = await serviceConnection.serviceManager.service!.getVM();
+    final isolates = vm.isolatesForDevToolsMode();
+    final vmDeveloperModeEnabled = preferences.vmDeveloperModeEnabled.value;
+    if (serviceManager.isolateManager.selectedIsolate.value!.isSystemIsolate! &&
+        !vmDeveloperModeEnabled) {
+      serviceManager.isolateManager
+          .selectIsolate(serviceManager.isolateManager.isolates.value.first);
+    }
+    await serviceManager.isolateManager.init(isolates);
   }
 
-  @override
-  List<IsolateRef> isolatesFromVm(VM? vm) {
-    return vm?.isolatesForDevToolsMode() ?? <IsolateRef>[];
+  Future<String?> rootLibraryForMainIsolate() async {
```

Comment:
this seems like something an extension might want to know. how do you draw the line on whether to include it in the shared code or not?

Reply:
Agreed we may want to expose this. For now, I'd like to keep the API small and add more to `devtools_app_shared` as it makes sense. At this point I am exposing everything needed for the package:provider extension as a customer zero, and will expose more as we get a better idea of what more initial customers need.